frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-topology/FinallyNode.tsx

```diff
@@ -0,0 +1,131 @@
+  const allTasksLength = finallyTasks.length + finallyListTasks.length;
+  return (
+    <Layer id="groups">
+      <>
+        <rect
+          className="odc-finally-node"
+          key={'rect'}
+          fill="transparent"
+          strokeWidth={1}
+          stroke="var(--pf-global--BorderColor--light-100)"
+          width={NODE_WIDTH + FINALLY_NODE_PADDING * 2}
+          height={
+            NODE_HEIGHT +
+            (allTasksLength - 1) * FINALLY_TASK_NODE_HEIGHT +
+            (!disableBuilder ? NODE_HEIGHT : 0) +
+            FINALLY_NODE_PADDING * 2
+          }
+          rx="20"
+          ry="20"
+        />
+
+        {finallyTasks.map((ft, i) => (
+          <g
+            key={ft.name}
+            className="odc-finally-builder-node"
+            transform={`translate(${FINALLY_NODE_PADDING}, ${FINALLY_TASK_NODE_HEIGHT * i +
+              FINALLY_NODE_PADDING})`}
+          >
+            <rect width={NODE_WIDTH} height={NODE_HEIGHT} fill="transparent" />
+            <g onClick={ft.onTaskSelection && !disableBuilder ? () => ft.onTaskSelection() : null}>
+              <PipelineVisualizationTask
+                pipelineRunName={pipelineRun?.metadata?.name}
+                task={ft}
+                pipelineRunStatus={pipelineRun && pipelineRunFilterReducer(pipelineRun)}
+                namespace={pipeline?.metadata?.namespace}
+                disableTooltip={!disableBuilder}
+                selected={ft.selected}
+                width={NODE_WIDTH}
+                height={NODE_HEIGHT}
+              />
+              {ft.error && !disableBuilder && (
+                <ErrorNodeDecorator
+                  x={BUILDER_NODE_ADD_RADIUS / 2}
+                  y={BUILDER_NODE_ERROR_RADIUS / 4}
+                  errorStr={ft.error}
+                />
+              )}
+            </g>
+          </g>
+        ))}
+        {!disableBuilder &&
+          finallyListTasks.map((flt, i) => (
+            <g
+              key={flt.name}
+              transform={`translate(${FINALLY_NODE_PADDING}, ${FINALLY_TASK_NODE_HEIGHT *
+                (i + finallyTasks.length) +
+                FINALLY_NODE_PADDING})`}
+            >
+              <TaskList
+                width={NODE_WIDTH}
+                height={NODE_HEIGHT}
+                listOptions={[...clusterTaskList, ...namespaceTaskList]}
+                onRemoveTask={flt.onRemoveTask}
+                onNewTask={flt.convertList}
+              />
+            </g>
+          ))}
+        {!disableBuilder && (
+          <g
+            transform={`translate(${(NODE_WIDTH + FINALLY_NODE_PADDING * 2) / 2}, ${allTasksLength *
+              FINALLY_TASK_NODE_HEIGHT +
+              FINALLY_ADD_LINK_TEXT_HEIGHT +
+              FINALLY_NODE_PADDING})`}
+          >
+            <PlusNodeDecorator
+              x={(FINALLY_NODE_PADDING + FINALLY_ADD_LINK_TEXT_HEIGHT + FINALLY_ADD_LINK_SIZE) * -1}
```

Comment:
`FINALLY_ADD_LINK_TEXT_HEIGHT` in a horizontal calculation? I'm confused. Height is a vertical construct and rarely does it make sense in a horizontal calculation. Can I get a little help with understanding?